### DISCUSSION TITLE
I would like to know the beacon default sleep time and timeout when team server is not found.

### DISCUSSION THREAD

**Discussion by pentester2222** [gloxec/CrossC2#203] [Q&A] [answered]
If I generate a beacon with its default setting for example like this: 
genCrossC2.Linux IP PORT <cobaltstrike.beacon_keys> null Linux x64 /tmp/t_cc.out

What would be its default timeout setting to connec to the team server again? For example if beacon is running but no team server is responding, how many times does it retries to connect? Lets say if beacon is not be able to contact the team server for a week or so long duruation, will that beacon be exited automatically? If that exists/stops after certain number of retries, how can i make it retry to connect to team server upto unlimited time untill i stop the process manually? 

Can anyone please help me related to this timeout and sleep setting?  I know I can set the sleep <time in seconds> as i test it is reconnecting to the team server to after specified seconds, if i set 1 hours time as a sleep timing, how many retires does it do to connect to team server?

if anyone can reply and explain on this it would be appreciated and it would help me alot for my testing and learning :) 
Thank you!

**Comment by gloxec** [comment, marked as answer]
`beacon` judgment connected to `TeamServer` is calculated by successfully establishing a communication port. For example, if you use CDN or Nginx, etc., even if the `TeamServer` is closed, `beacon` has successfully established a communication with the setting `C2 HOST`.

Under the above judgment rules:
1. After multiple communication fails, the request frequency is changed to one hour once, and the original frequency will not be restored until successfully established
2. After continuous communication fails, after the `sleep` does not interrupt more than **48h**, `beacon` will automatically exit
---
3. You can use a custom `void cc2_retryConnect(int retryCount) {}` function hook built -in error heavy connection stage. Its use method is similar to [custom-protocol-dynamic-library](https://github.com/gloxec/CrossC2/wiki/genCrossC2#2-use-a-custom-protocol-dynamic-library)
4. If you use (3. custom-dynamic-library), you can use the configuration of `config.ini` to modify the symbol of the hook function parsing to prevent reverse analysis
https://github.com/gloxec/CrossC2/blob/43171fc134964b2586bef7c5989fa37cea6d5778/src/config_demo.ini#L6-L7
